src/Kriti/Parser/Token.hs

```diff
@@ -0,0 +1,147 @@
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+{-# LANGUAGE RankNTypes #-}
+module Kriti.Parser.Token where
+
+import qualified Data.HashMap.Strict as M
+import qualified Data.List as L
+import Data.Scientific (Scientific)
+import qualified Data.Text as T
+import qualified Data.Vector as V
+import GHC.Generics
+import Kriti.Parser.Spans
+
+data Symbol =
+    SymBling
+  | SymColon
+  | SymDot
+  | SymComma
+  | SymEq
+  | SymGt
+  | SymLt
+  | SymAnd
+  | SymOr
+  | SymSingleQuote
+  | SymCurlyOpen
+  | SymCurlyClose
+  | SymDoubleCurlyOpen
+  | SymDoubleCurlyClose
+  | SymSquareOpen
+  | SymSquareClose
+  | SymParenOpen
+  | SymParenClose
+  | SymUnderscore
+  | SymAssignment
+  | SymStringBegin 
+  | SymStringEnd 
+  deriving (Show, Eq, Ord, Generic)
+
+data Token =
+    TokSymbol (Loc Symbol)
+  | TokStringLit (Loc T.Text)
+  | TokIdentifier (Loc T.Text)
+  | TokNumLit T.Text (Loc Scientific)
+  | TokIntLit T.Text (Loc Int)
+  | TokBoolLit (Loc Bool)
+  | EOF
+  deriving (Show, Eq, Ord, Generic)
+
+overLoc :: (forall a. Loc a -> Loc a) -> Token -> Token
+overLoc f (TokSymbol loc) = TokSymbol $ f loc
+overLoc f (TokStringLit loc) = TokStringLit $ f loc
+overLoc f (TokIdentifier loc) = TokIdentifier $ f loc
+overLoc f (TokNumLit txt loc) = TokNumLit txt $ f loc
+overLoc f (TokIntLit txt loc) = TokIntLit txt $ f loc
+overLoc f (TokBoolLit loc) = TokBoolLit $ f loc
+overLoc _ EOF = EOF
+
+serialize :: Token -> T.Text
```

Comment:
This function is used strictly for rending parse errors and should be replaced by a span aware pretty printer in subsequent PR.

Reply:
This is probably worth a `-- NOTE:` comment stating as much, or something.